sympy/functions/elementary/hyperbolic.py

```diff
@@ -209,6 +209,52 @@ class cosh(HyperbolicFunction):
     sinh, tanh, acosh
     """
 
+    @property
+    def is_positive(self):
+        from sympy import pi
+        arg = self.args[0]
+
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+        if im % pi == 0 or re == 0:
```

Comment:
Yes, that can be done, but I don't think there would be a big performance difference due to that. I think the current way works fine as well.